openvino/conditional_compilation/include/openvino/cc/factory.h

```diff
@@ -0,0 +1,144 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+#include "selective_build.h"
+#include <string>
+#include <functional>
+#include <type_traits>
+#include <unordered_map>
+
+namespace openvino
```

Comment:
I have a question about the code style.
What code style do you use?

Because it is not aligned with IE (google) and nGraph (llvm) styles.

I think we need to align it and enable code style check for this library.

Reply:
Code style checkers are defined inside IE. I've tried to avoid adding a common libraries dependency on IE.

Reply:
See https://github.com/openvinotoolkit/openvino/pull/3228

Reply:
I suppose we can use common coding style after merge of #3228. I propose to merge coding style changes for all common libraries (itt and cc) in other PR.

Reply:
I agree

Reply:
@vladislav-volkov please, feel free to continue with code style